dsp-shared/src/main/scala/dsp/valueobjects/Iri.scala

```diff
@@ -36,6 +50,157 @@ object Iri {
+   *
+   * @param iri the IRI to be checked.
+   */
+  def isKnoraGroupIriStr(iri: IRI): Boolean =
+    Iri.isIri(iri) && iri.startsWith("http://rdfh.ch/groups/")
+
+  /**
+   * Returns `true` if an IRI string looks like a Knora project IRI
+   *
+   * @param iri the IRI to be checked.
+   */
+  def isKnoraProjectIriStr(iri: IRI): Boolean =
+    (iri.startsWith("http://rdfh.ch/projects/") || isKnoraBuiltInProjectIriStr(iri))
+
+  /**
+   * Returns `true` if an IRI string looks like a Knora built-in IRI:
+   *  - http://www.knora.org/ontology/knora-admin#SystemProject
+   *  - http://www.knora.org/ontology/knora-admin#SharedOntologiesProject
+   *
+   * @param iri the IRI to be checked.
+   */
+  def isKnoraBuiltInProjectIriStr(iri: IRI): Boolean = {
+
+    val builtInProjects = Seq(
+      SystemProject,
+      DefaultSharedOntologiesProject
+    )
+
+    Iri.isIri(iri) && builtInProjects.contains(iri)
+  }
+
+  // Characters that are escaped in strings that will be used in SPARQL.
+  private val SparqlEscapeInput = Array(
+    "\\",
+    "\"",
+    "'",
+    "\t",
+    "\n"
+  )
+
+  // Escaped characters as they are used in SPARQL.
+  private val SparqlEscapeOutput = Array(
+    "\\\\",
+    "\\\"",
+    "\\'",
+    "\\t",
+    "\\n"
+  )
+
+  /**
+   * Makes a string safe to be entered in the triplestore by escaping special chars.
+   *
+   * If the param `revert` is set to `true`, the string is unescaped.
+   *
+   * @param s        a string.
+   * @param errorFun a function that throws an exception. It will be called if the string is empty or contains
+   *                 a carriage return (`\r`).
+   * @return the same string, escaped or unescaped as requested.
+   */
+  def toSparqlEncodedString(s: String, errorFun: => Nothing): String = {
+// Findings about this method:
+// - there is more cases to handle according to docs (https://www.w3.org/TR/rdf-sparql-query#grammarEscapes) - we use 1.1 version right?
+// - `'` doesn't appear on that list, but this method escapes it
+// - why `\r` throws error instead of being escaped?
+// - fun fact is that if I remove  StringUtils.replaceEach, for example `\t` passes unescaped, why?
+
+    if (s.isEmpty || s.contains("\r")) errorFun
+
+    // http://www.morelab.deusto.es/code_injection/
+
+    StringUtils.replaceEach(
+      s,
+      SparqlEscapeInput,
+      SparqlEscapeOutput
+    )
+  }
+
+  /**
+   * Checks that a string represents a valid IRI. Also encodes the IRI, preserving existing %-escapes.
+   *
+   * @param s        the string to be checked.
+   * @param errorFun a function that throws an exception. It will be called if the string does not represent a valid
+   *                 IRI.
+   * @return the same string.
+   */
+  def validateAndEscapeIri(s: String, errorFun: => Nothing): IRI = {
+    val urlEncodedStr = encodeAllowEscapes(s)
+
+    if (Iri.urlValidator.isValid(urlEncodedStr)) {
+      urlEncodedStr
+    } else {
+      errorFun
```

Comment:
one thing to note for a later PR: the `v2.scala` is duplicating an older state of StringFormatter, so there may be improvements there, that are not yet included in here.  
When you/we harmonize the value objects and StringFormatter, we have to make sure to include all the changes that were made recently.

Reply:
Thanks for the reminder. I will put this as a note into next step task.